go/datas/pull.go

```diff
@@ -19,7 +19,10 @@ type PullProgress struct {
 	DoneCount, KnownCount, ApproxWrittenBytes uint64
 }
 
-const bytesWrittenSampleRate = .10
+const (
+	bytesWrittenSampleRate = .10
+	batchSize              = 1 << 15 // 32k chunks
```

Comment:
does this mean 128MB average batch size (assuming 4k chunks)? Did you try adjusting this up & down to see the impact on perf. I might have guessed a lower value (e.g. 10s of MB) would have been optimal.

Reply:
It does mean that, yeah. I can play around with it a little to see if anything seems immediately better

Reply:
I dropped it all the way down to ~16MB per batch (4096 chunks at avg 4k per) before the performance started to slowly drop off